src/backend/opencl/compile_module.cpp

```diff
@@ -119,6 +122,16 @@ Program buildProgram(const vector<string> &kernelSources,
 
 }  // namespace opencl
 
+string getKernelCacheFilename(const int device, const string &key) {
+    auto &dev = opencl::getDevice(device);
+
+    unsigned vendorId = dev.getInfo<CL_DEVICE_VENDOR_ID>();
+    auto devName      = dev.getInfo<CL_DEVICE_NAME>();
```

Comment:
Perhaps remove spaces or replace with underscores from the device name? Some file systems may interpret paths with spaces differently.

Another thing to consider is to use the same case.